File: unit4/LVCustomAdapater/app/src/main/java/com/example/lvcustomadapater/MyAdapter.java
package com.example.lvcustomadapater;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;


import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class MyAdapter extends BaseAdapter {
    Context context;
    ArrayList<Restuarant> arr;
    public MyAdapter(Context context, ArrayList<Restuarant> arr) {
        this.context=context;
        this.arr= arr;

    }

    @Override
    public int getCount() {
        return arr.size();
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        LayoutInflater inflater=  LayoutInflater.from(context);
        view= inflater.inflate(R.layout.list_item,null);

        TextView name= view.findViewById(R.id.name);
        TextView rating= view.findViewById(R.id.rating);
        ImageView imag= view.findViewById(R.id.imageView);
        ImageView loc= view.findViewById(R.id.locition);

        name.setText(arr.get(i).name);
        rating.setText(arr.get(i).rating);
        imag.setImageResource(arr.get(i).imag);
        loc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent= new Intent(Intent.ACTION_VIEW, Uri.parse(arr.get(i).loc));
                context.startActivity(intent);
            }
        });
        return view;
    }
}

File: Kotlin_Projects/ResturantApp/app/src/main/java/com/example/resturantapp/MyAdapter.kt
package com.example.resturantapp

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView

class MyAdapter(var context: Context, var Arr: ArrayList<Resturant>) : BaseAdapter() {


    override fun getCount(): Int {
        return Arr.size
    }

    override fun getItem(p0: Int): Any {
        TODO("Not yet implemented")
    }

    override fun getItemId(p0: Int): Long {
        return p0.toLong()

    }

    override fun getView(p0: Int, p1: View?, p2: ViewGroup?): View {
val view:View= View.inflate(context, R.layout.list_item, null)

         var name: TextView = view.findViewById(R.id.name)
        var rating: TextView = view.findViewById(R.id.rating)
        var img: ImageView= view.findViewById(R.id.imageView)
        var loc: ImageView = view.findViewById(R.id.locition)
        name.text= Arr.get(p0).name
        rating.text=Arr.get(p0).rating
        img.setImageResource(Arr.get(p0).img)

loc.setOnClickListener(View.OnClickListener {
val i= Intent(Intent.ACTION_VIEW, Uri.parse(Arr.get(p0).location))
    context.startActivity(i)
})
return view
    }
}